### PR TITLE
Auth service

### PR DESCRIPTION
**Pull request by grusha229**
MVP of authentication service implemented.

### CODE REVIEW

**Comment by vr009** [comment]
Don't forget about:

- [ ] swagger

- [ ] unit tests

- [ ] integration tests (postman will be ok for that purpose)

Github actions will be added later, after all.

File: internal/auth/internal/delivery/handler.go [hunk 1]
```diff
@@ -2,94 +2,171 @@ package delivery
...
-		log.Print(err)
-	} else {
-		w.Header().Set("Set-Cookie", fmt.Sprintf("ssid=%s; path=/; HttpOnly", token))
-		http.Redirect(w, r, "/", http.StatusSeeOther)
+	if errCode != models.OK {
+		checkErrorCode(errCode, w)
+		return
 	}
+
+	w.WriteHeader(http.StatusOK)
```
Comment (vr009) [inline internal/auth/internal/delivery/handler.go:120]: You have another const value - model.Deleted. Please, refactor your logic for using this value (looks not much to change in code).

File: internal/auth/internal/models/session.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import (
+	"github.com/google/uuid"
+	"time"
+)
+
+type Session struct {
+	UserID    uuid.UUID `json:"user_id"`
+	Token     string    `json:"token,omitempty"`
+	ExpiredAt time.Time `json:"expired_at,omitempty"`
```
Comment (vr009) [inline internal/auth/internal/models/session.go:11]: This field seems unused. It is needed to keep in mind that tokens must be expired one day. We need to use it in redis and here, in code. Fix it after all other fixes, please.

File: internal/auth/internal/repo/db.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+//redis-server
+
+func NewDataBase(Client *redis.Client) *DataBase {
+	return &DataBase{
+		Client: Client,
+	}
+}
+
+func (db *DataBase) CreateSession(session models.Session) (models.Session, models.StatusCode) {
+	token, err := uuid.NewUUID()
```
Comment (vr009) [inline internal/auth/internal/repo/db.go:25]: Well, uuid is actually not the best format for keeping tokens. We should look at some other solutions for generate tokens. 
At least, our requirements for the implementation:
- it should be a fixed length string (or slice of bytes)
- it must be generated fast

I have found this [tread](https://stackoverflow.com/questions/22892120/how-to-generate-a-random-string-of-a-fixed-length-in-go/22892986#22892986)

File: internal/auth/internal/repo/db.go [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+	value, err := cmd.Result()
+	if err != nil {
+		return models.Session{}, models.InternalError
+	}
+
+	return models.Session{
+		Token:  value,
+		UserID: session.UserID,
+	}, models.OK
+
```
Comment (vr009) [inline internal/auth/internal/repo/db.go:67]: Extra line, let's remove it.

**Review by vr009 — CHANGES_REQUESTED** [review]

File: internal/auth/internal/repo/db.go [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+	uuidFormString, err := uuid.Parse(value)
+
+	if err != nil {
+		log.Print("error of UUID parsing")
+		return models.Session{}, models.InternalError
+	}
+
+	fmt.Println(value)
+	fmt.Println(uuidFormString)
```
[the remark this answers is not in the document]
Comment (grusha229) [reply internal/auth/internal/repo/db.go:95]: done

File: internal/auth/internal/delivery/handler.go (hunk 1 above)
Comment (grusha229) [reply internal/auth/internal/delivery/handler.go:120]: done

File: internal/auth/internal/repo/db.go (hunk 3 above)
Comment (grusha229) [reply internal/auth/internal/repo/db.go:25]: done

File: internal/auth/internal/repo/db.go (hunk 4 above)
Comment (grusha229) [reply internal/auth/internal/repo/db.go:67]: done

File: internal/auth/internal/repo/db.go (hunk 3 above)
Comment (vr009) [reply internal/auth/internal/repo/db.go]: We should also test if the collision is possible here. The benchmark for 100500 times of generating tokens fits here good. To be honestly, we could just calculate a hash from users id. Use [crypto package](https://pkg.go.dev/crypto) for this purpose.

File: internal/auth/internal/repo/db.go [hunk 6]
```diff
@@ -3,51 +3,48 @@ package repo
...
 		Client: Client,
+		TTL:    TTL,
 	}
 }
 
 func (db *DataBase) CreateSession(session models.Session) (models.Session, models.StatusCode) {
-	token := createToken(64)
-
-	//nowTime := time.Now()
+	token := MD5(session.UserID.String())
```
Comment (vr009) [inline internal/auth/internal/repo/db.go:32]: That should be moved to usecase layer.

**Comment by vr009** [comment]
You have commited artifacts from your IDE, that files must not be here. Also you need to remove user_data service code.
You should remove it, squash commits and force push it here.

**Comment by vr009** [comment]
The PR needs to be rebased on `main` now.

File: internal/auth/internal/repo/db.go (hunk 6 above)
Comment (vr009) [reply internal/auth/internal/repo/db.go:32]: Will be moved to issue as your tech debt.

**Review by vr009 — APPROVED** [review]
Tests, swagger and other issues will be moved to your tech debts.

**Pull request merged**